app/models/secret.rb

```diff
@@ -0,0 +1,12 @@
+class Secret < ActiveRecord::Base
+  belongs_to :sender, class_name: "User"
+  belongs_to :receiver, class_name: "User"
+  belongs_to :message
+  belongs_to :algorithm
+
+  validates :sender, :receiver, :message, :algorithm, presence: true
+
+  def pretty_date
+    self.created_at.strftime("%B %-d, %Y")
```

Comment:
I'd make this a helper, or add to the rails standard date format set by using an initializer - it's usually better not to pollute your models with display-only concerns.